modules/ossm-exposing-a-gateway-to-traffic-outside-the-cluster-using-openshift-routes.adoc

```diff
@@ -0,0 +1,58 @@
+// This procedure is used in the following assembly:
+// * gateways/ossm-getting-traffic-into-a-mesh.adoc 
+
+:_mod-docs-content-type: PROCEDURE
+[id="ossm-exposing-a-gateway-to-traffic-outside-the-cluster-using-openshift-routes_{context}"]
+= Exposing a gateway to traffic outside the cluster by using OpenShift Routes
+:context: ossm-exposing-a-gateway-to-traffic-outside-the-cluster-using-openshift-routes
+
+You can expose a gateway to traffic outside the cluster by using {ocp-short-name} Routes. This approach provides an alternative to using Kubernetes load balancer service when you have to expose gateways to traffic outside the cluster.
+
+.Prerequisites
+
+* You have installed a gateway using gateway injection.
+
+.Procedure
+
+. Ensure that the `Service`` type is set to `ClusterIP`.
++
+[source,terminal]
+----
+oc patch service  -n  -p '{"spec": {"type": "ClusterIP"}}'
+----
+
+. Create a `Route`.
++
+[source,yaml,subs="attributes,verbatim"]
+----
+apiVersion: route.openshift.io/v1
+kind: Route
+metadata:
+  name: httpbin
+  namespace: <gateway_namespace>
+spec:
+  host: httpbin.example.com
+  port:
+    targetPort: http2
+  to:
+    kind: Service
+    name: <gateway_name>
+    weight: 100
+  wildcardPolicy: None
+  # ...
+----
+
+. Verify that httpbin service can be accessed from outside the cluster through the ingress router. Ensure that you set the `INGRESS_HOST` variable appropriately for the environment that your cluster is running in. 
++
+[source,terminal]
+----
+# If the cluster is running on AWS:
+INGRESS_HOST=$(oc get service router-default -n openshift-ingress -o jsonpath='{.status.loadBalancer.ingress[0].hostname}')
+
+# If the cluster is running on GCP or Azure:
+INGRESS_HOST=$(oc get service router-default -n openshift-ingress -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+
+curl -s -I -H Host:httpbin.example.com http://$INGRESS_HOST/headers
+----
```

Comment:
These three commands should be in separate code blocks and need a `$` prompt.
```suggestion
[source,terminal]
----
# If the cluster is running on AWS:
$ INGRESS_HOST=$(oc get service router-default -n openshift-ingress -o jsonpath='{.status.loadBalancer.ingress[0].hostname}')
----
+
[source,terminal]
----
# If the cluster is running on GCP or Azure:
$ INGRESS_HOST=$(oc get service router-default -n openshift-ingress -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
----
+
[source,terminal]
----
$ curl -s -I -H Host:httpbin.example.com http://$INGRESS_HOST/headers
----
```

Reply:
@mburke5678  This will be addressed in https://issues.redhat.com/browse/OSSM-8338 which is already underway